examples/collector/src/Makefile

```diff
@@ -0,0 +1,69 @@
+# http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+#
+# Copyright 2016 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+PLUGINTYPE=collector
+NAME=rando
+TARGET=test-$(PLUGINTYPE)-$(NAME)
+
+SNAPLIBS=/usr/local/lib
+ICCROOT=
+EROOT=../..
+BDIR=$(EROOT)/bin
+
+CC=g++
+IDIRS=-I.
+LDIRS=-L.
+CFLAGS=-std=c++1y
+LIBS=-Wl,--start-group $(SNAPLIBS)/libsnap.a $(SNAPLIBS)/libgrpc.a $(SNAPLIBS)/libgrpc++.a $(SNAPLIBS)/libprotobuf.a -Wl,--end-group -lz -lpthread
```

Comment:
I've tested in clear environment, steps:
```
$ vagrant init ubuntu/xenial64; vagrant up --provider virtualbox
$ vagrant ssh
$ sudo apt-get install make autoconf automake libtool gettext g++ curl
$ git clone https://github.com/grpc/grpc
$ cd grpc/
$ git checkout tags/v1.0.1
$ git submodule update --init
$ make
$ sudo make install
$ cd third_party/protobuf/
$ sudo make install
$ cd $HOME
$ git clone https://github.com/intelsdi-x/snap-plugin-lib-cpp.git
$ snap-plugin-lib-cpp/
$ git fetch origin pull/32/head:pull_request_32
$ git checkout -b pull_request_32
$ ./autogen.sh
$ ./configure
$ make
$ sudo make install
$ cd examples/
$ make
```
Output :
```
make[1]: Entering directory '/home/ubuntu/snap-plugin-lib-cpp/examples/collector/src'
g++ -std=c++1y -I. -L. -o test-collector-rando rando.cc -Wl,--start-group /usr/local/lib/libsnap.a /usr/local/lib/libgrpc.a /usr/local/lib/libgrpc++.a /usr/local/lib/libprotobuf.a -Wl,--end-group -lz -lpthread
/usr/bin/ld: cannot find -lz
collect2: error: ld returned 1 exit status
Makefile:69: recipe for target 'test-collector-rando' failed
make[1]: *** [test-collector-rando] Error 1
make[1]: Leaving directory '/home/ubuntu/snap-plugin-lib-cpp/examples/collector/src'
 
make[1]: Entering directory '/home/ubuntu/snap-plugin-lib-cpp/examples/processor/src'
g++ -std=c++1y -I. -L. -o test-processor-graffiti graffiti.cc -Wl,--start-group /usr/local/lib/libsnap.a /usr/local/lib/libgrpc.a /usr/local/lib/libgrpc++.a /usr/local/lib/libprotobuf.a -Wl,--end-group -lz -lpthread
/usr/bin/ld: cannot find -lz
collect2: error: ld returned 1 exit status
Makefile:69: recipe for target 'test-processor-graffiti' failed
make[1]: *** [test-processor-graffiti] Error 1
make[1]: Leaving directory '/home/ubuntu/snap-plugin-lib-cpp/examples/processor/src'
 
make[1]: Entering directory '/home/ubuntu/snap-plugin-lib-cpp/examples/publisher/src'
g++ -std=c++1y -I. -L. -o test-publisher-log log.cc -Wl,--start-group /usr/local/lib/libsnap.a /usr/local/lib/libgrpc.a /usr/local/lib/libgrpc++.a /usr/local/lib/libprotobuf.a -Wl,--end-group -lz -lpthread
/usr/bin/ld: cannot find -lz
collect2: error: ld returned 1 exit status
Makefile:69: recipe for target 'test-publisher-log' failed
make[1]: *** [test-publisher-log] Error 1
make[1]: Leaving directory '/home/ubuntu/snap-plugin-lib-cpp/examples/publisher/src'
```
but I need to install zlib1g-dev (`sudo apt-get install zlib1g-dev`) and then it works.